File: python/patterns/multiple_pointers.py
# sumZero receives an array of numbers and return the first pair where sums is zero
def sumZero(numbers):
    # if the len is zero should return None
    if (len(numbers) == 0):
        return None

    # create two pointer for the start and end of the array
    left = 0
    right = len(numbers) - 1

    # interate until left is less than right pointer
    while (left < right):
        sum = numbers[left] + numbers[right]
        # if the sum of the pointers value is equal zero return the pointers
        if (sum == 0):
            return [numbers[left], numbers[right]]
        # if the sum is greater than zero decrease right pointer
        elif (sum > 0):
            right -= 1
        # otherwise increase the left pointer
        else:
            left += 1

    # if the pair is not found return None
    return None

File: go/patterns/multiple_pointers.go
package patterns

// SumZero should return the first pair where sum is equal to zero, otherwise nil
func SumZero(numbers []int) []int {
	// if the len is equal to zero return nil
	if len(numbers) == 0 {
		return nil
	}

	// create two pointers for the start and end of the array
	left := 0
	right := len(numbers) - 1

	// interate with the numbers arrays
	for left < right {
		// get the sum of the two pointers values
		sum := numbers[left] + numbers[right]
		// if the sum is equal to zero return the pointers values
		if sum == 0 {
			return []int{numbers[left], numbers[right]}
			// if the sum is greater than zero decrease right pointer
		} else if sum > 0 {
			right--
			// otherwise increase the left pointer
		} else {
			left++
		}
	}

	// otherwise, return nil
	return nil
}
